waspc/src/Wasp/Analyzer/Parser/SourceSpan.hs

```diff
@@ -0,0 +1,20 @@
+{-# LANGUAGE DeriveGeneric #-}
```

Comment:
Also added this.

So before we had SourcePosition and SourceRegion.
Now we also have SourceOffset and SourceSpan.

SourceRegion is really a tuple of SourcePositions.
SourceSpan is really a tuple of SourceOffsets.
SourcePosition is 2D (line, column).
SourceOffset is 1D (offset).

We already had both offset and span in the code already, they just didn't have any types (no type for offset, and type called Region for Span, which was confusing), so now they have nice names, and are nicely positioned.

Reply:
I agree, names are not clearly indicative right in the sense is it 1D or 2D.
I tried coming up with those, but it wasn't super clear what works the best.

For example, adding Linear into the name sounds good, I could do smth like this:
SourcePosition, SourceRegion, SourceLinearPosition, SourceLinearRegion.
Then it is clear what is the relation between these, in all the ways.
Hm I actually wonder why I didn't use this, this is actually pretty good :D.

I think I didn't because I liked term "offset" since it is pretty precise. Problem is then finding a name for "offset region".
I could have: SourcePosition, SourceRegion, SourceOffset, SourceOffsetRegion, but it felt a bit weird due to SourceRegion not being SourcePositionRegion.

Btw I was trying to not rename SourcePosition or SourceRegion since they are used on many places already, and I actually do like the names.

If we went with: SourcePosition, SourceRegion, LinearSourceOffset, LinearSourceSpan -> I think here it is not very obvious what the relations are.

I ended up going with SourceOffset and SourceSpan basically giving up on any obvious relation, but at least I thought there are also no wrong implications and you will just have to learn what means what.

I am not sure at the end. One better option does seem to be the first one that I mentioned: SourcePosition, SourceRegion, SourceLinearPosition, SourceLinearRegion.
The new names are a bit longish though. A bit easier to see "offset" in code, then "linearPosition" or "linearRegion". Although the second is more explicit.

Reply:
Yeah, all good points. And it really could be just me that was hoping for some additional disambiguation keyword to differentiate 1d vs 2d with the original terms. But maybe the Linear versions are not necessary. I'd say we can see if @sodic likes the original, and if so, let's just use that 👍🏻 

And makes sense trying to keep the original terms given the amount of refactoring if those changed.

Reply:
Thanks @shayneczyzewski  -> ok let's see what @sodic says, if needed I can do the renaming!

Reply:
We can leave them as they are as far as I'm concerned. Out of curiosity, why do we have both approaches in our codebase?

Reply:
In ConcreteParser, and in CST, it structure, offset is heavily used (and therefore also span). Basically 1D stuff. I believe this was most natural for the problem itself while @craigmc08 was solving it.